From 5411c6322bf51896b57ebf0f29aab561513e413d Mon Sep 17 00:00:00 2001
From: Nadav Kavalerchik <nadavkav@gmail.com>
Date: Fri, 28 Sep 2018 16:19:55 +0300
Subject: [PATCH 90/95] mod/assign (patch) Disable grader link

---
 mod/assign/gradingtable.php | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/mod/assign/gradingtable.php b/mod/assign/gradingtable.php
index 542f573..c558aa9 100644
--- a/mod/assign/gradingtable.php
+++ b/mod/assign/gradingtable.php
@@ -938,7 +938,7 @@ class assign_grading_table extends table_sql implements renderable {
         if (!$this->is_downloading() && $this->hasgrade) {
             $urlparams = array('id' => $this->assignment->get_course_module()->id,
                                'rownum' => 0,
-                               'action' => 'grade'); // disable "grader" link (nadavkav)
+                               'action' => 'grader'); // disable "grader" link (nadavkav)
 
             if ($this->assignment->is_blind_marking()) {
                 if (empty($row->recordid)) {
@@ -1175,7 +1175,7 @@ class assign_grading_table extends table_sql implements renderable {
 
         $urlparams = array('id' => $this->assignment->get_course_module()->id,
                                'rownum' => 0,
-                               'action' => 'grade'); // disable "grader" link (nadavkav)
+                               'action' => 'grader'); // disable "grader" link (nadavkav)
 
         if ($this->assignment->is_blind_marking()) {
             if (empty($row->recordid)) {
-- 
1.8.3.1

