From 4491d856b9c285a7ac5df475ba3e4a5aaa781a0c Mon Sep 17 00:00:00 2001
From: Nadav Kavalerchik <nadavkav@gmail.com>
Date: Fri, 28 Sep 2018 21:55:34 +0300
Subject: [PATCH 92/95] Sort Nav drawer courses by last access

---
 lib/navigationlib.php | 21 ++++++++++++++++++++-
 1 file changed, 20 insertions(+), 1 deletion(-)

diff --git a/lib/navigationlib.php b/lib/navigationlib.php
index a915894..e6981a8 100644
--- a/lib/navigationlib.php
+++ b/lib/navigationlib.php
@@ -2906,7 +2906,7 @@ class global_navigation extends navigation_node {
      * They've expanded the 'my courses' branch.
      */
     protected function load_courses_enrolled() {
-        global $CFG;
+        global $CFG, $USER, $DB;
 
         $limit = (int) $CFG->navcourselimit;
 
@@ -2917,9 +2917,28 @@ class global_navigation extends navigation_node {
         }
         // Append the chosen sortorder.
         $sortorder = $sortorder . ',' . $CFG->navsortmycoursessort . ' ASC';
+
+        // Special sorting of idnumber desc
+        if (!empty($CFG->navsortmycoursessort) &&
+            ($CFG->navsortmycoursessort === 'idnumberdesc' || $CFG->navsortmycoursessort === 'lastaccess')) {
+            $sortorder = 'visible DESC, idnumber DESC';
+        }
+
         $courses = enrol_get_my_courses('*', $sortorder);
         $flatnavcourses = [];
 
+        // Sort course by user's lastaccess.
+        if (!empty($CFG->navsortmycoursessort) && $CFG->navsortmycoursessort === 'lastaccess') {
+            $lastaccesscourses = $DB->get_records('user_lastaccess', array('userid' => $USER->id), 'timeaccess DESC');
+            foreach ($lastaccesscourses as $c) {
+                if (isset($courses[$c->courseid])) {
+                    $courses[$c->courseid]->lastaccess = $c->timeaccess;
+                }
+            }
+            // Sort by user's lastaccess to course
+            usort($courses, function($a, $b) { return $b->lastaccess - $a->lastaccess; });
+        }
+
         // Go through the courses and see which ones we want to display in the flatnav.
         foreach ($courses as $course) {
             $classify = course_classify_for_timeline($course);
-- 
1.8.3.1

